### DISCUSSION TITLE
Feature Request: Add as parameter to router.push in /app structure for client-side URL masking

### DISCUSSION THREAD

**Discussion by Nandeep2750** [vercel/next.js#69608] [App Router]
In the next js `/pages` structure, there is an option for mask URLs for the browser using `router.push`

https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#caveats

```
import { useRouter } from 'next/router';
const router = useRouter();

// node_modules/next/dist/shared/lib/router/router.d.ts

/**
   * Performs a `replaceState` with arguments
   * @param url of the route
   * @param as masks `url` for the browser
   * @param options object you can define `shallow` and other options
*/
push(url: Url, as?: Url, options?: TransitionOptions): Promise<boolean>;

```

So If I want to use this same feature in next `/app` structure so how we can use that? as in `import { useRouter } from 'next/navigation'` that option is now available.

```
import { useRouter } from 'next/navigation';
const router = useRouter();

// node_modules/next/dist/shared/lib/app-router-context.shared-runtime.d.ts
 /**
   * Navigate to the provided href.
   * Pushes a new history entry.
*/
push(href: string, options?: NavigateOptions): void;
```

**Comment by Juneezee** [comment]
This was asked before in https://github.com/vercel/next.js/discussions/48110.

You can use the native `window.history.pushState` and `window.history.replaceState`, see https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#using-the-native-history-api
